juriscraper/opinions/united_states/federal_special/tax.py

```diff
@@ -93,56 +92,132 @@ def _get_download_urls(self):
         for td in self.html.xpath(path):
             date_string = td.text_content().strip()
-            dates.append(datetime.strptime(date_string, '%m/%d/%Y').date())
+            dates.append(datetime.strptime(date_string, "%m/%d/%Y").date())
         return dates
 
     def _download_backwards(self, d):
         self.backwards_days = 7
         self.case_date = d
-        logger.info("Running backscraper with date range: %s to %s",
-                    self.case_date - timedelta(days=self.backwards_days),
-                    self.case_date)
+        logger.info(
+            "Running backscraper with date range: %s to %s",
+            self.case_date - timedelta(days=self.backwards_days),
+            self.case_date,
+        )
         self.html = self._download()
         if self.html is not None:
             # Setting status is important because it prevents the download
             # function from being run a second time by the parse method.
             self.status = 200
+
+    def extract_from_text(self, opinion_text):
+        """Can we extract the citation and related information
+
+        :param opinion_text: The content of the docuemnt downloaded
+        :return: dictionary of values
+        """
+
+        metadata = {}
+
+        tax_court_reports_regex = re.compile(
+            r"""
+            ([0-9]{1,4})\s{1,}                           # (volume)
+            UNITED\ STATES\ TAX\ COURT\ REPORTS?\s{1,}   # (reporter)
+            \((\d{1,4})\)                                # (page)
+            """,
+            re.VERBOSE | re.IGNORECASE,
+        )
+
+        tax_court_alt_regex = re.compile(
+            r"""
+            ((T\.\ ?C\.\s((Memo\.?)|(Summary\ Opinion))\s{1,}   # T.C. Memo or Summary Opinion (reporter)
+            ([0-9]{4})                                          # Four digit year (volume)
+            .                                                   # hyphen, em-dash etc.
+            ([0-9]{1,3})\b)                                     # 1-3 digit number in order of publication (page)
+            |                                                   # or 
+            ([0-9]{1,4})\s{1,}                                  # (volume)
+            (T\.\ ?C\.\ No\.)\s{1,}                             # T.C. No. (reporter)
+            (\d{1,4}))                                          # (page)
+            """,
+            re.VERBOSE | re.IGNORECASE,
+        )
+
+        match = re.search(tax_court_reports_regex, opinion_text)
+
+        if match:
+            metadata["cite"] = match.group()
+            metadata["volume"] = match.group(1)
+            metadata["page"] = match.group(2)
+            metadata["precedential_status"] = "published"
+            metadata["reporter"] = "T.C."
+            metadata["normalized_citation"] = "%s %s %s" % (
+                metadata["volume"],
+                metadata["reporter"],
+                metadata["page"],
+            )
```

Comment:
Is it crazy here to do what you did for LASC? You had some magic code you wrote that could take a dict and make joined objects, didn't you? At the very least, should the dict keys here mirror our data model so something similar is possible in CL?

Reply:
I dont follow exactly

Reply:
I'm just trying to think about the CL side of things. If we get a dict, how do we figure out which pieces of the data in the dict get merged with which pieces of the CL data model. More to the point, if we get back a `cite` key, how does CL know what to do with it. Is there a way to be clever about this so that CL can save things correctly according to their key names? Didn't we have that in place at LASC somewhere? There's a weird block of code with maybe a while loop and some key parsing that does some funky work to import scraped dicts —- no?

Reply:
Ah. Okay that makes more sense.  We did have a function that mapped directly to the model and because of that could loop and store data automatically.

Reply:
Latest push should match returned object to model - and tests reflect that